samples/main.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+    "io/ioutil"
+    "log"
+    "path/filepath"
+    "os"
+    "os/exec"
+    "runtime"
+)
+
+func main() {
+    _, filename, _, _ := runtime.Caller(0)
+    root := filepath.Dir(filename)
+    dirs, err := ioutil.ReadDir(root)
+    if err != nil {
+        log.Fatal(err)
+    }
+
+    for _, d := range dirs {
+        testFile := filepath.Join(root, d.Name(), "test.go")
+        if _, err := os.Stat(testFile); err == nil {
+            cmd := exec.Command("go", "run", testFile)
```

Comment:
Do we want to run these in parallel?

Reply:
Yes, we'd better run them in parallel.

Reply:
I am guessing these will also be replaced by `go test <test-proj>` ?